core/test/utils/matrix_utils_test.cpp

```diff
@@ -57,70 +57,138 @@ class MatrixUtils : public ::testing::Test {
     using value_type = T;
     using real_type = gko::remove_complex<T>;
     using mtx_type = gko::matrix::Dense<T>;
+    using mtx_data = gko::matrix_data<value_type, int>;
 
     MatrixUtils()
         : exec(gko::ReferenceExecutor::create()),
-          mtx(gko::test::generate_random_matrix<mtx_type>(
+          data(gko::test::generate_random_matrix_data<value_type, int>(
               500, 500, std::normal_distribution<real_type>(50, 5),
               std::normal_distribution<real_type>(20.0, 5.0),
-              std::default_random_engine(42), exec)),
-          unsquare_mtx(mtx_type::create(exec, gko::dim<2>(500, 100)))
+              std::default_random_engine(42))),
+          rectangular_data(gko::dim<2>(500, 100))
     {}
 
     std::shared_ptr<const gko::Executor> exec;
-    std::unique_ptr<mtx_type> mtx;
-    std::unique_ptr<mtx_type> unsquare_mtx;
+    mtx_data data;
+    mtx_data rectangular_data;
 };
 
 TYPED_TEST_SUITE(MatrixUtils, gko::test::ValueTypes, TypenameNameGenerator);
 
 
 TYPED_TEST(MatrixUtils, MakeSymmetricThrowsError)
 {
-    ASSERT_THROW(gko::test::make_symmetric(gko::lend(this->unsquare_mtx)),
+    ASSERT_THROW(gko::test::make_symmetric(this->rectangular_data),
                  gko::DimensionMismatch);
 }
 
 TYPED_TEST(MatrixUtils, MakeHermitianThrowsError)
 {
-    ASSERT_THROW(gko::test::make_hermitian(gko::lend(this->unsquare_mtx)),
+    ASSERT_THROW(gko::test::make_hermitian(this->rectangular_data),
                  gko::DimensionMismatch);
 }
 
 
 TYPED_TEST(MatrixUtils, MakeDiagDominantThrowsError)
 {
-    ASSERT_THROW(gko::test::make_diag_dominant(gko::lend(this->mtx), 0.9),
+    ASSERT_THROW(gko::test::make_diag_dominant(this->data, 0.9),
                  gko::ValueMismatch);
 }
 
 
 TYPED_TEST(MatrixUtils, MakeHpdMatrixThrowsError)
 {
-    ASSERT_THROW(gko::test::make_hpd(gko::lend(this->mtx), 1.0),
-                 gko::ValueMismatch);
+    ASSERT_THROW(gko::test::make_hpd(this->data, 1.0), gko::ValueMismatch);
+}
+
+
+TYPED_TEST(MatrixUtils, MakeLowerTriangularCorrectly)
+{
+    auto orig_mtx = TestFixture::mtx_type::create(this->exec);
+    orig_mtx->read(this->data);
+    gko::test::make_lower_triangular(this->data);
+
+    auto mtx = TestFixture::mtx_type::create(this->exec);
+    mtx->read(this->data);
+    for (gko::size_type i = 0; i < mtx->get_size()[0]; i++) {
+        for (gko::size_type j = 0; j <= i; j++) {
+            ASSERT_EQ(mtx->at(i, j), orig_mtx->at(i, j));
+        }
+        for (gko::size_type j = i + 1; j < mtx->get_size()[1]; j++) {
+            ASSERT_EQ(mtx->at(i, j), gko::zero<TypeParam>());
```

Comment:
Should we stop at the first different number? Or use an EXPECT to go through all of them?
```suggestion
            EXPECT_EQ(mtx->at(i, j), orig_mtx->at(i, j));
        }
        for (gko::size_type j = i + 1; j < mtx->get_size()[1]; j++) {
            EXPECT_EQ(mtx->at(i, j), gko::zero<TypeParam>());
```

Reply:
I guess the only reason I used ASSERT is because the output would be huge in the failure case otherwise.